source/MaterialXGenMdl/MdlShaderGenerator.cpp

```diff
@@ -431,4 +431,45 @@ namespace MDL
     const string OUTPUTS  = "o";
 }
 
+bool MdlShaderGenerator::remapEnumeration(const ValueElement& input, const string& value, std::pair<const TypeDesc*, ValuePtr>& result) const
+{
+    // Early out if not an enum input.
+    const string& enumNames = input.getAttribute(ValueElement::ENUM_ATTRIBUTE);
+    if (enumNames.empty())
+    {
+        return false;
+    }
+
+    // Don't convert or filenames and arrays.
+    const TypeDesc* type = TypeDesc::get(input.getType());
+    if (type == Type::FILENAME || type->isArray())
+    {
+        return false;
+    }
+
+    // Try remapping to an enum value.
+    if (!value.empty())
+    {
+        MdlSyntaxPtr mdlSyntax = std::dynamic_pointer_cast<MdlSyntax>(_syntax);
+        result.first = mdlSyntax->getEnumeratedType(value);
```

Comment:
Set enumerated type here. but keep the value as a string which will be handled by custom syntax.